src/commands/show.ts

```diff
@@ -47,17 +47,24 @@ export async function searchProblem(): Promise<void> {
     await showProblemInternal(choice.value);
 }
 
-export async function showSolution(node?: LeetCodeNode): Promise<void> {
-    if (!node) {
+export async function showSolution(input: LeetCodeNode | vscode.Uri): Promise<void> {
+    let problemInput: string | undefined;
+    if (input instanceof LeetCodeNode) {
+        problemInput = input.id;
+    } else if (input instanceof vscode.Uri) {
+        problemInput = `"${input.fsPath}"`;
+    } else {
+        vscode.window.showErrorMessage("Invalid input to fetch the solution data");
         return;
     }
+
     const language: string | undefined = await fetchProblemLanguage();
```

Comment:
We could utilize file extension (`path.extName()`) and `shared.ts/langExt` to determine the language, thus no need for re-selecting the language when showing solution from the code:
```ts
export const langExt: Map<string, string> = new Map([
    ["bash", "sh"],
    ["c", "c"],
    ["cpp", "cpp"],
    ["csharp", "cs"],
    ["golang", "go"],
    ["java", "java"],
    ["javascript", "js"],
    ["kotlin", "kt"],
    ["mysql", "sql"],
    ["php", "php"],
    ["python", "py"],
    ["python3", "py"],
    ["ruby", "rb"],
    ["rust", "rs"],
    ["scala", "scala"],
    ["swift", "swift"],
]);
```
(Yet this choice comes with a risk: currently showing solution for languages like scala has a high probability  to fail. This feature could be added in later PR when logic of solution plugin is improved.)

Reply:
another rick is that it's hard to differentiate py3 & py2 from the `extName`